utils/src/logging.rs

```diff
@@ -21,3 +21,8 @@ pub fn setup_logging() -> Result<()> {
 
     Ok(())
 }
+
+pub fn gen_request_id() -> u16 {
+    // For now, it's just a random value. In future, we are going to get this value depending on context
+    rand::random()
+}
```

Comment:
I think this is a bad idea. We should _not_ use content based IDs, since TXs can be retried, and re-using the same ID makes debugging harder.

We should also no generated our own IDs, but use something like open temelemetry:

https://docs.rs/opentelemetry/0.21.0/opentelemetry/trace/struct.SpanContext.html#method.trace_id

Reply:
As far as I can tell, this is not really used in this PR? If so, I'd remove this from here and we can implement it in a subsequent PR.

Reply:
I found only one usage: https://github.com/0xPolygonMiden/miden-node/pull/160/files#diff-78024d37013859b8564cd8f586d2bbd875dfa0c2200acf4c5346ce4d24e85d6bR43

Reply:
I see. Maybe still remove it from there and do a more comprehensive handling of requests in a different PR.

Reply:
@hackaugusto 

> I think this is a bad idea. We should _not_ use content based IDs, since TXs can be retried, and re-using the same ID makes debugging harder.

This is an opposite, not a content based ID, but unique number for tracing of parallel calls. Unfortunately, there is no native support of such tracing in `tracing-subscriber` crate, so I tried to implement this feature (we discussed with @bobbinth this yesterday). Maybe we should find another subscriber, which has support of tracing nested calls within a single request?

Reply:
> This is an opposite, not a content based ID, but unique number for tracing of parallel calls

Ah, thank you, I misunderstood the comment.

> Unfortunately, there is no native support of such tracing in tracing-subscriber crate, so I tried to implement this feature (we discussed with @bobbinth this yesterday). Maybe we should find another subscriber, which has support of tracing nested calls within a single request?

Just to make sure we are in the same page:

- The current subscriber already traces calls in a single request. And here single request means one request inside one server, not a distributed request. In the [screenshot above](https://github.com/0xPolygonMiden/miden-node/pull/160#discussion_r1458033740), the white colored text, after the level `INFO` represent this tracing:

```
serve:
serve:load:
serve:load:load_nullifier_tree:
serve:load:load_mmr:
serve:load:load_accounts
```

The trace above means that `load_nullifier_tree`, `load_mmr`, and `load_accounts` are run by `load`, which itself in turn is run inside `serve`, the top-down order is execution order. That tracing is done using spans' relationships, parent/child + sibling. The subscriber main purpose is to generate unique span ids for the above tracing. (important caveat, these ids are reusable)

- The above is for a request in a single server. IMO we should also have distributed tracing, across a cluster. For this, we first have to decide which distributed tracer to use, for example opentelemetry, and then we can see if we can reuse the ID generated by that.

This [piece of docs](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/registry/struct.Registry.html#span-id-generation) is relevant for this discussion.

Reply:
@igamigo I'm new to `tracing` lib and maybe I'm wrong, but what if we have several concurrent queries, will it separate them in output or we will have a mishmash from different thread outputs?

Reply:
> will it separate them in output or we will have a mishmash from different thread outputs?

The lines would be interwinded.

> How to separate them?

You need to correlate the lines.

Just to clarify. I'm in favor of having IDs, the point I was trying to make is that these IDs have to be threaded through different servers. It is not sufficient to have local unique IDs.